html_app/components/MicroscopyModelFactory.js

```diff
@@ -2,31 +2,33 @@ scb.components = scb.components || {};
 
 scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelFactory(model, template) {
     var self = this;
-    
-    if(scb.utils.isDefined(model.slide)){
+
+    if(scb.utils.isDefined(model.slide)){//model here is template.model.microscopy
     	self.slide = function(state){
-    			
-            	var t = template;
-            	var m = model.slide;
-//     			console.log('state');
-//     			console.log(state);
-    			if (m.parser_simple){
-    				var hash_list = []; 
-    				_.each(state.microscopy.lanes_list.list, function(x){
-							if(x.current_slides.length > 0){
-								_.each(x.current_slides, function(slide){
-									hash_list.push(slide.hash);
-								});
-							}
-			
-						});
-					var microscopy_lane = state.microscopy_lane;
-					var cell_line = microscopy_lane.cell_treatment.cell_line;
-					var collection_id= microscopy_lane.cell_treatment.treatment_list.first.collection_id;
-					var drug_id = microscopy_lane.cell_treatment.treatment_list.first.drug_list.list[0].drug_id;
-					var slide_type = microscopy_lane.kind;	
-					var conditions = microscopy_lane.slide_conditions;
-					var imgs = []
+            var hash_list = [];
+            _.each(state.microscopy.lanes_list.list, function(x){
+                if(x.current_slides.length > 0){
+                    _.each(x.current_slides, function(slide){
+                    	hash_list.push(slide.hash);
+                    });
+                }
+            });
+
+            var microscopy_lane = state.microscopy_lane;
+            var cell_line = microscopy_lane.cell_treatment.cell_line;
+            var collection_id= microscopy_lane.cell_treatment.treatment_list.first.collection_id;
+            var drug_id = microscopy_lane.cell_treatment.treatment_list.first.drug_list.list[0].drug_id;
+            var slide_type = microscopy_lane.kind;
+            var conditions = microscopy_lane.slide_conditions;
+            var imgs = []
+            var isFound = false;
+            var max;
+            var index;
+            var m = model.slide;
+
+            if (m.parser_simple){
```

Comment:
the assignment should have this attribute (or one of the other two: complex_parser or condition_parser)